scipy/stats/contingency.py

```diff
@@ -273,3 +275,124 @@ def chi2_contingency(observed, correction=True, lambda_=None):
                                    lambda_=lambda_)
 
     return chi2, p, dof, expected
+
+
+def association(n_obs, n_rows, n_cols, chi2_stat, stat="V"):
```

Comment:
Wouldn't it be better to have this function simply take `observed` as a single parameter? Looking at the tests, all these parameters are derived from a single set of observations, so why not save the user 4 lines and calculate them inside the function?

@josef-pkt made the same comment in an earlier review.

Reply:
So, that was one of the changes I made in this latest PR. The main reason I removed it was because there was a lot of contention over how to calculate the stat for an Nd matrix say for one that is 4d like (2x2x5x3). The way I had it was that it would produce an output that was 5x3 matrix of association stats (in this case). 

I am willing to put it back but honestly it seemed like it was much less hassle this way, and it can always be expanded to suit that purpose in a future PR.  

Another solve is to set the function to reject anything that isn't 2d.

Anyway, let me know what you think I should do.

Reply:
Okay, I missed that part of the discussion. I think the concerns of @josef-pkt and @WarrenWeckesser are for >2-D arrays, and when you change to `n_rows, `n_cols` you're anyway assuming 2-D only right? So that would be easier to achieve by keeping `observed` but simply raising an error for >2-D input.

Let's see what the other reviewers say.